### ISSUE TITLE
[ORM] SearchContext not returning correct summary for fields that search associated objects

### ISSUE THREAD

**Issue by colintucker** [silverstripe/silverstripe-framework#7778]
## Affected Version

4.0.1

## Description

When using `ModelAdmin`, a `DataObject` may specify `$searchable_fields` that filter by a property on an associated object, for example:

```php
$searchable_fields = [
    'Address.Country' => [
        'filter' => 'ExactMatchFilter',
        'field' => CountryDropdownField::class
    ]
];
```

This would show a dropdown to filter objects with an associated address object that has the specified country.  The filtering works perfectly fine, however fields of this type (i.e. using the "dot syntax") do not appear to show up in the `$SearchSummary` that appears at the top of `ModelAdmin`.

I believe the issue lies within the `getSummary()` method in `SearchContext`, and is connected to the renaming of "dot syntax" fields from using the period character to double-underscore, i.e. `Address.Country` becomes `Address__Country`.

Even though the filtering itself works, I believe this issue should be fixed primarily for UX concerns, i.e. a user may become confused as to whether the filtering is working or not without the search summary at the top of their model admin.

## Steps to Reproduce

1. Make a `DataObject` with a `$has_one` to another `DataObject`.
2. Add a "dot syntax" field to `$searchable_fields` for the first `DataObject` that filters by a property on the associated `DataObject`
3. Make a `ModelAdmin` for the objects.
4. Attempt to filter using the "dot syntax" field.
5. The filtered field should not currently appear in the search summary.

**Comment by unclecheese** [comment]
This issue appears to have been fixed by the [rebuild of the search UI](https://github.com/silverstripe/silverstripe-cms/issues/1777). Can you double check if it's still an issue, @colintucker ?

**Comment by maxime-rainville** [comment]
That got fixed with this issue https://github.com/silverstripe/silverstripe-admin/issues/672.

The fix only got applied to the 4.3 branch however.

**Comment by wernerkrauss** [comment]
Still an issue or fixed in 4.3+?

**Comment by maxime-rainville** [comment]
Everything from 4.3.0 onwards will have the fix.
